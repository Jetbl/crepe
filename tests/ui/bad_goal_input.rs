// In this test, the relation `In` is inappropriately derived in a rule, even
// though it is marked as an @input.

mod datalog {
    use crepe::crepe;

    crepe! {
        @input
        struct In(u32, u32);

        In(0, 0) <- (true);
    }
}

fn main() {}
