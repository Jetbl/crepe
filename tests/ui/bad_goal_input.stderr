error: Relations marked as @input cannot be derived from a rule.
  --> $DIR/bad_goal_input.rs:11:9
   |
11 |         In(0, 0) <- (true);
   |         ^^
